lib/parse.js

```diff
@@ -20,9 +20,13 @@ function parse(input) {
   let shouldDecode = false;
   let hasPlus = false;
   const lastIndex = input.length;
+  const il = lastIndex + 1;
+  let i = 0;
+  let type = "";
+  let currentValue;
 
   // Have a boundary of input.length + 1 to access last pair inside the loop.
-  for (let i = 0; i < lastIndex + 1; i++) {
+  for (; i < il; ++i) {
```

Comment:
In terms of maintenance & readability, this worse things. What about performance? What's the impact of this change?